src/view_neume.cpp

```diff
@@ -93,249 +97,176 @@ void View::DrawNc(DeviceContext *dc, LayerElement *element, Layer *layer, Staff
-{
-    assert(dc);
-    assert(layer);
-    assert(staff);
-    assert(measure);
-
-    Neume *neume = dynamic_cast<Neume *>(element);
+    // Determine grouping component is a part of and its position
+    Neume *neume = dynamic_cast<Neume *>(nc->GetParent());
     assert(neume);
 
-    /******************************************************************/
-    // Initialization
+    NeumeGroup group = neume->GetNeumeGroup();
+    int position = neume->GetChildIndex(element);     
 
-    // Obtaining list of NC to extract intm data
-    // Format of *neumechildren, first object is <neume>, all objects after are <nc>
-    const ListOfObjects neumeChildren = neume->GetList(neume)[0];
-    if (neumeChildren.empty()) {
-        return;
-    }
+    std::vector<int> pitchDifferences = neume->GetPitchDifferences();
 
-    std::vector<Nc *> ncVector;
-    std::vector<int> pitchDifferences;
-    for (int i = 1; i < (int)neumeChildren.size(); i++) {
-        Nc *nc = dynamic_cast<Nc *>(neumeChildren[i]);
-        ncVector.push_back(nc);
-        if (i >= 2) {
-            // Calculate the pitch differences between each nc pair
-            Nc *prev_nc = dynamic_cast<Nc *>(neumeChildren[i - 1]);
-            int pitchDifference = nc->GetPname() - prev_nc->GetPname();
-            pitchDifference += (nc->GetOct() - prev_nc->GetOct()) * 7;
-            pitchDifferences.push_back(pitchDifference);
-        }
-    }
-
-    // Obtaining the NeumeGroup enum name from the map
-    NeumeGroup neumeName = neume->GetNeumeGroup();
-
-    // If the shape cannot be found in the map, NeumeGroup::ERROR will be returned since its
-    // value is 0 and std::map returns 0 by default for missing keys.
-    if (neumeName == NEUME_ERROR) {
-        // TODO: Error Handling for unfound neumes
-        LogError("Neume Grouping not found");
-    }
-
-    /******************************************************************/
-    // Start the Neume graphic and draw the children
-
-    dc->StartGraphic(element, "", element->GetUuid());
-    // If there is a defined way to draw the neume grouping, we should use that to draw the specialized SVG shape,
-    // or else we draw each nc in the neume as a punctum.
-
-    //// THIS DOESNT WORK YET BECAUSE GLYPHS NOT IN LEIPZIG
-    // int glyphSize = m_doc->GetGlyphWidth(SMUFL_E990_chantPunctum, staff->m_drawingStaffSize, 0);
-    int noteHeight = (int)(m_doc->GetDrawingDoubleUnit(staff->m_drawingStaffSize) / 2);
-    int noteWidth = (int)(m_doc->GetDrawingDoubleUnit(staff->m_drawingStaffSize) / 1.4);
-    switch (neumeName) {
+    switch (group) {
         case PES:
-        case PES_SUBPUNCTIS: 
+        case PES_SUBPUNCTIS:
         {
-            int xOffset = 0;
-            if (pitchDifferences.at(0) > 1) {
-                xOffset = -1;
+            if (position == 1 && pitchDifferences.at(0) > 0) {
+                params.at(0).xOffset = -0.25;

```

Comment:
What unit is this? percentage? If not you should express is as DrawingUnit

Reply:
It's as a percentage of the height/width of a note for `yOffset` and `xOffset` respectively.